src/DynamoCore/Core/PathManager.cs

```diff
@@ -436,6 +427,23 @@ private string GetDynamoDataFolder(string folder)
                 String.Format("{0}.{1}", majorFileVersion, minorFileVersion));
         }
 
+        private string TransformPath(string root, string extension)
+        {
+            if (root.StartsWith(GetUserDataFolder()))
+                return Path.Combine(root, extension);
+            try
+            {
+                var subFolder = Path.Combine(root, extension);
+                if (!Directory.Exists(subFolder))
+                    return subFolder;
+            }
+            catch (IOException) { }
+            catch (ArgumentException) { }
+            catch (UnauthorizedAccessException) { }
+
+            return root;
+        }
+
```

Comment:
More documentation for this method (what goes in, what comes out, examples, etc.).